### PR TITLE
Separate distance from co2 calculation

### PR DESCRIPTION
**Pull request by akokam**

### CODE REVIEW

File: co2calculator/calculate.py [hunk 1]
```diff
@@ -587,13 +588,17 @@ def calc_co2_businesstrip(
...
         if transportation_mode == "car" and (
             type(start) != str or type(destination) != str
         ):
-            raise ValueError(
-                "Wrong data type for start and destination."
-                "Please provide a three letter IATA code for train stations."
-            )
+            # NOTE: I turned off that type check since lower level functions expect dicts
+            # (calc_co2_car & geo_coding_structured both want dictionaries)
+            pass
```
Comment (akokam) [inline co2calculator/calculate.py:597]: @han16nah Last git changes are referencing you :)

I had to turn off this check to make my functional test work. I was wondering though if I missed something.

My analysis showed:

In the event of `calc_co2_businesstrip()` with `transportation_mode = "car"`, it checks for type str for `start` and `destination`.

BUT: Both lower-level functions - `calc_co2_car` and `geocode_structured` - require a dictionary.

It actually works with disabling the checks (see `tests/functional/test_calculate.py`.


My question is: is `calculate_co2_businesstrip` used atm and if .. does is work? 😂

Comment (han16nah) [reply co2calculator/calculate.py:597]: Huh, it seems like something is very messed up here.
I guess "car" would need to be replaced with "plane" because that's where a string with the IATA code is needed.. 
https://github.com/pledge4future/co2calculator/blob/67592506a59431e3cbee1645d8a52196efd4e0b4/co2calculator/calculate.py#L333
And I don't know what's up with the Error message then :D this should obviously be plane as well

Should I fix something and push to dev or will you just make the changes in your branch?

PS: Nah, probably does not work ;)

Comment (akokam) [reply co2calculator/calculate.py:597]: How about this: I finish functional tests first. If necessary, comment and fix things. If that's all done, I need you (or someone else) to review if the test data is what the backend really sends. 

If the tests are set up, I gonna apply changes to distance calculation and input validation anyway!

**Comment by akokam** [comment]
@han16nah, I finished the distance-based and stop-based functional testing for the `calc_co2_businesstrip()` interface. Github actions are failing due to API Key error (still need to look into it; or .. better introduce mocking external calls).

However, as I mentioned in slack: A few changes to `calculate.py` were necessary to make the code run (1: Disable the type checking for `start` and `stop`; 2: Fix unpacking of coordinates).

Can you please:
1) Check if my tests look somewhat like your planned input from backend (looping in @redfrexx)
2) Maybe double check the emissions .. they seem to be a little upside down sometimes .. (not too important right now)

Thanks!

File: tests/functional/test_calculate.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+                    "locality": "Heidelberg",
+                    "country": "Germany",
+                },
+                {
+                    "country": "Germany",
+                    "locality": "Berlin",
+                    "address": "Alexanderplatz 1",
+                },
+                # NOTE: This seems slightly unrealistic.
+                # TODO: Please check business logic (can be different branch)!
```
Comment (han16nah) [inline tests/functional/test_calculate.py:69]: Seems like I found the bug, please check #73 :)

**Review by redfrexx — APPROVED** [review]
Hey, sorry for the delay, but looks very good so far! :) Thanks a lot!

**Comment by akokam** [comment]
Sorry, it's my first time starting in an existing project. I've decided that the branch would have been too big. Hence, I separated into two main tasks: (1, #79) pin down the interfaces to backend & (2, coming) changing the code to really separate the distance form co2 calculation!

tl;dr: closed in favor of #79
